src/lib.rs

```diff
@@ -12,127 +12,201 @@
-//! let number: u32 = bytes.pread::<u32>(0).unwrap();
-//! // ...or a byte, with type ascription on the binding.
-//! let byte: u8 = bytes.pread(0).unwrap();
 //!
-//! //If the type is known another way by the compiler, say reading into a struct field, we can omit the turbofish, and type ascription altogether!
+//! // Similarly we can also read a single byte at offset 2
+//! // This time using type ascription instead of the turbofish (::<>) operator.
+//! let byte: u8 = bytes.pread(2).unwrap();
+//! assert_eq!(byte, 0xbe);
 //!
-//! // If we want, we can explicitly add a endianness to read with by calling `pread_with`.
-//! // The following reads a u32 out of `b` with Big Endian byte order, at offset 0
+//!
+//! // If required we can also provide a specific parsing context; e.g. if we want to explicitly
+//! // define the endianess to use:
 //! let be_number: u32 = bytes.pread_with(0, scroll::BE).unwrap();
-//! // or a u16 - specify the type either on the variable or with the beloved turbofish
-//! let be_number2 = bytes.pread_with::<u16>(2, scroll::BE).unwrap();
+//! assert_eq!(be_number, 0xdeadbeef);
+//!
+//! let be_number16 = bytes.pread_with::<u16>(1, scroll::BE).unwrap();
+//! assert_eq!(be_number16, 0xadbe);
+//!
 //!
-//! // Scroll has core friendly errors (no allocation). This will have the type `scroll::Error::BadOffset` because it tried to read beyond the bound
-//! let byte: scroll::Result<i64> = bytes.pread(0);
+//! // Reads may fail; in this example due to a too large read for the given container.
+//! // Scroll's error type does not by default allocate to work in environments like no_std.
+//! let byte_err: scroll::Result<i64> = bytes.pread(0);
+//! assert!(byte_err.is_err());
 //!
-//! // Scroll is extensible: as long as the type implements `TryWithCtx`, then you can read your type out of the byte array!
 //!
-//! // We can parse out custom datatypes, or types with lifetimes
-//! // if they implement the conversion trait `TryFromCtx`; here we parse a C-style \0 delimited &str (safely)
-//! let hello: &[u8] = b"hello_world\0more words";
+//! // We can parse out custom datatypes, or types with lifetimes, as long as they implement
+//! // the conversion traits `TryFromCtx/FromCtx`.
+//! // Here we use the default context for &str which parses are C-style '\0'-delimited string.
+//! let hello: &[u8] = b"hello world\0more words";
 //! let hello_world: &str = hello.pread(0).unwrap();
-//! assert_eq!("hello_world", hello_world);
+//! assert_eq!("hello world", hello_world);
 //!
-//! // ... and this parses the string if its space separated!
-//! use scroll::ctx::*;
-//! let spaces: &[u8] = b"hello world some junk";
-//! let world: &str = spaces.pread_with(6, StrCtx::Delimiter(SPACE)).unwrap();
-//! assert_eq!("world", world);
+//! // We can again provide a custom context; for example to parse Space-delimited strings.
+//! let hello2: &[u8] = b"hello world\0more words";
```

Comment:
hmm, this example confused me at first, but i see now it illustrates both space delimited parsing, and the null terminator is correctly seen as single item, so it's good

Reply:
Yeah reading it a few days after writing it it's not so obvious what I'm  trying to demonstrate - I hope the expanded comment helps with clarity.